src/main/webapp/app/overview/exercise-details/exercise-details-student-actions.component.html

```diff
@@ -139,6 +139,12 @@
                                 <span class="d-none d-md-inline" jhiTranslate="artemisApp.exerciseActions.openRepository"></span>
                             </a>
                         }
+                        @if (theiaEnabled) {
+                            <a class="btn btn-primary" [class.btn-sm]="smallButtons" (click)="startOnlineIDE()" target="_blank" rel="noopener noreferrer">
+                                <fa-icon [icon]="faDesktop" [fixedWidth]="true" />
+                                <span class="d-none d-md-inline" jhiTranslate="artemisApp.exerciseActions.openOnlineIDE"></span>
+                            </a>
```

Comment:
Since there seems to have quite a bit of future work planned that has to be added to the component, and `exercise-details-student-actions` is already quite long and does a lot of stuff: What do you think about moving this into a small dedicated component?

Reply:
We plan to keep the client-side of this (in fact pretty large) feature very lean, most functionality will be pushed in the server via the REST API. Thus, I'd leave it like this for now :)